mbedtls/src/x509/certificate.rs

```diff
@@ -153,16 +77,47 @@ fn x509_buf_to_vec(buf: &x509_buf) -> Vec<u8> {
     slice.to_owned()
 }
 
-fn x509_time_to_time(tm: &x509_time) -> Result<super::Time> {
+fn x509_time_to_time(tm: &x509_time) -> Result<Time> {
     // ensure casts don't underflow
     if tm.year < 0 || tm.mon < 0 || tm.day < 0 || tm.hour < 0 || tm.min < 0 || tm.sec < 0 {
         return Err(Error::X509InvalidDate);
     }
 
-    super::Time::new(tm.year as u16, tm.mon as u8, tm.day as u8, tm.hour as u8, tm.min as u8, tm.sec as u8).ok_or(Error::X509InvalidDate)
+    Time::new(tm.year as u16, tm.mon as u8, tm.day as u8, tm.hour as u8, tm.min as u8, tm.sec as u8).ok_or(Error::X509InvalidDate)
 }
 
-impl LinkedCertificate {
+
+impl Certificate {
+    pub fn from_der(der: &[u8]) -> Result<MbedtlsBox<Certificate>> {
+        let cert = MbedtlsBox::<Certificate>::new()?;
+        unsafe { x509_crt_parse_der(cert.as_ref().into(), der.as_ptr(), der.len()) }.into_result()?;
+        Ok(cert)
+    }
+
+    pub fn from_pem(pem_or_der: &[u8]) -> Result<MbedtlsList<Certificate>> {
```

Comment:
`from_pem` should check that there is only one certificate and then add `from_pem_multiple` that returns a list